File: code_samples/java/TestingMocking.java
/**
 * Testing: Mocking
 * AI/ML Training Sample
 */
public class TestingMocking {
    
    private String data;
    
    public TestingMocking() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        TestingMocking instance = new TestingMocking();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}

File: code_samples/elixir/testing_mocking.ex
# Testing: Mocking
# AI/ML Training Sample

defmodule Mocking do
  defstruct data: ""
  
  def new(), do: %Mocking{}
  
  def process(%Mocking{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Mocking{data: data}), do: data
  
  def validate(%Mocking{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Mocking.new()
updated = Mocking.process(instance, "example")
IO.puts("Data: " <> Mocking.get_data(updated))
IO.puts("Valid: " <> to_string(Mocking.validate(updated)))
